### PR TITLE
Single post view interaction refactorings

### LINKED ISSUE #6314: Don't automatically load likers and resharers profile images if over X amount for post
See picture:

11836869_10207752369516894_3820456825363901969_n

The only thing that does is kill the browser when accessing a popular post in the SVP :) Maybe just show X amount and then some 'show X more' link that would fetch the rest.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25682200-don-t-automatically-load-likers-and-resharers-profile-images-if-over-x-amount-for-post?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by svbergerem**
Fixes #6314. Related to #7168.

This loads the first 30 likes and reshares via gon and adds a 'Show all' link if there are more. Clicking on the link will fetch all likes or reshares.

show all likes

Comments are still fetched via Ajax.

### CODE REVIEW

**Comment by svbergerem** [comment]
There was a small conflict with #7419: the timeago in the single post view didn't work anymore. I fixed that while rebasing the PR.

**Review by cmrd-senya — CHANGES_REQUESTED** [review]
I didn't manage to review all changes of the PR, only a part of them. I'll continue with it in a few days.

File: app/assets/javascripts/app/views/single-post-viewer/single_post_interaction_counts.js [hunk 1]
```diff
@@ -2,10 +2,21 @@
 
 app.views.SinglePostInteractionCounts = app.views.Base.extend({
   templateName: "single-post-viewer/single-post-interaction-counts",
-  tooltipSelector: ".avatar.micro",
+  tooltipSelector: [
+    ".avatar.micro",
+    "#show-all-likes",
+    "#show-all-comments"
```
Comment (svbergerem) [inline app/assets/javascripts/app/views/single-post-viewer/single_post_interaction_counts.js]: Thanks! There is no need for a tooltip, so I removed the lines.

File: spec/presenters/post_presenter_spec.rb [hunk 2]
```diff
@@ -20,6 +20,26 @@
...
+
+    it "works without a user" do
+      expect(@unauthenticated_presenter.with_interactions).to be_a Hash
+    end
+  end
+
+  describe "#with_initial_interactions" do
+    it "works with a user" do
+      expect(@presenter.with_initial_interactions).to be_a Hash
+    end
```
Comment (cmrd-senya) [inline spec/presenters/post_presenter_spec.rb]: I think these tests could be a little smarter and verify the contents of the hash. This data isn't checked anywhere now.

**Review by cmrd-senya — APPROVED** [review]
Looks good to me, but here are some notes on the code

File: app/assets/javascripts/app/router.js [hunk 3]
```diff
@@ -182,7 +182,7 @@ app.Router = Backbone.Router.extend({
   },
 
   singlePost: function(id) {
-    this.renderPage(function() { return new app.pages.SinglePostViewer({id: id}); });
+    this.renderPage(function() { return new app.pages.SinglePostViewer({id: id, el: $("#container")}); });
```
Comment (svbergerem) [inline app/assets/javascripts/app/router.js]: I was pretty sure that there is a reason for this but I also had to search for a while. Without this change the timeagos in the single post view are missing and only appear after a few seconds. That's the conflict I mentioned [here](https://github.com/diaspora/diaspora/pull/7182#issuecomment-301717784). I'll move this change to a new commit.

File: spec/presenters/post_presenter_spec.rb (hunk 2 above)
Comment (svbergerem) [inline spec/presenters/post_presenter_spec.rb]: Done.

**Review by skyschub — APPROVED** [review]

**Comment by denschub** [comment]
@svbergerem mind rebasing this?

**Comment by svbergerem** [comment]
@denschub Done.

**Comment by SuperTux88** [comment]
Merged (after 9 months ;) ), thanks everyone ❤️

**Pull request merged**